### DISCUSSION TITLE
Prevent updating a package not pinned by Renovate

### DISCUSSION THREAD

**Discussion by adamj88** [renovatebot/renovate#20210] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.118

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

.com EE

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

In my use case, renovate has updated a package to version 1.4.2 however, for compatibility we require it to be locked to version 1.3.1. 

After renovate has performed version pinning and we have downgraded it, renovate will then come back on the next run and upgrade it to 1.4.2, in other instances, it has been explicitly set to a version and then upgraded after renovate has done the initial pinning but not during.

Is there any configuration to prevent a package from being updated when its version has been explicitly set to a version? 

Ideally, I would like to achieve this without having to configure the package name instead, it is picked up from the dependencies automatically.

### What you tried so far.

Reading the documentation and searching through discussions

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Renovate should find the old merged MR and be "blocked" by it (if the new version is the same as the original PR).

i.e. if there's an old MR called "Update foo to v1.4.2" then you shouldn't see a new one the same.

**Reply by adamj88** [reply]
@rarkins what if renovate never ran before this and the version has been pre-pinned for the purpose of locking the version specifically to that one?

**Reply by rarkins** [reply]
If you never want Renovate to update that package, you need to introduce config in order to do so.

Something like this:

```json
{
  "packageRules": [{
    "matchCurrentValue": "/^\\d+\\.\\d+\\.\\d+$/",
    "enabled": false
  }]
}
```

In other words, disable any dependency where the value is like `1.2.3` and not e.g. `^1.2.3`
